File: app/src/main/java/com/left/gank/ui/base/fragment/SupportFragment.java
package com.left.gank.ui.base.fragment;

import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;

import com.left.gank.utils.ToastUtils;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

/**
 * Create by LingYan on 2017-09-28
 */

public abstract class SupportFragment extends ButterKnifeFragment {
    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
    }

    @Override
    public void shortToast(String str) {
        if (!TextUtils.isEmpty(str)) {
            ToastUtils.showToast(getContext(), str);
        }
    }
}

File: app/src/main/java/com/left/gank/base/fragment/SupportFragment.kt
package com.left.gank.base.fragment

import com.left.gank.utils.ToastUtils

/**
 * Create by LingYan on 2017-09-28
 */

abstract class SupportFragment : ButterKnifeFragment() {
    override fun shortToast(msg: String) {
        if (!isDetached) {
            ToastUtils.showToast(context, msg)
        }
    }
}
